Assets/HoloToolkit/UX/Scripts/Tooltip/ToolTipBackground.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using HoloToolkit.Unity;
+
+namespace HoloToolkit.UX.ToolTips
+{
+    [RequireComponent(typeof(MeshFilter))]
+
+    /// <summary>
+    /// Base class for a tool tip background
+    /// Automatically finds a ToolTip and subscribes to ContentChange action
+    /// Resizes its content to match ToolTip's content in ScaleToFitContent()
+    /// </summary>
+    public abstract class ToolTipBackground : MonoBehaviour
+    {
+        [SerializeField]
+        protected ToolTip toolTip;
+
+        protected virtual void OnEnable()
+        {
+            toolTip = gameObject.EnsureComponent<ToolTip>();
+
+            if (toolTip == null)
+            {
+                Debug.LogError("No tooltip found in ToolTipMeshBackground");
+                enabled = false;
+                return;
+            }
+            
+            toolTip.ContentChange += ContentChange;
+        }
+
+        protected virtual void ContentChange()
+        {
+            ScaleToFitContent();
+        }
+
+        protected abstract void ScaleToFitContent();
```

Comment:
should a default implementation be provided?

Reply:
the overrides (tooltipBackgroundMesh, tooltipBackgroundBlob, and tooltipBackgroundCorners have such different overrides for this abstract function that it really isn't possible to make a useful virtual function with default implementation. I could change the abstract prototype to a virtual function and have it be empty but that seems like it violates the thinking behind the original design.